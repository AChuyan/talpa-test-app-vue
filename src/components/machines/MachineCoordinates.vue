<template>
  <p class="machine-coordinates">
    <span>lat={{ get(coordinates, 'lat', placeholder) }}</span>
    <span>lng={{ get(coordinates, 'lng', placeholder) }}</span>
  </p>
</template>

<script>
  import get from "lodash.get";

  export default {
    data() {
      return {
        placeholder: 'unknown',
      }
    },
    methods: {
      get,
    },
    name: 'machine-coordinates',
    props: ['coordinates'],
  }
</script>

<style scoped>
  .machine-coordinates > span {
    font: 500 1rem 'Avenir', Helvetica, Arial, sans-serif;
  }
  .machine-coordinates > span:first-of-type {
    padding-right: .5rem;
  }
</style>
