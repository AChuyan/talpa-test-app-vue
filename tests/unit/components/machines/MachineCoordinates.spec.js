import { shallowMount } from '@vue/test-utils'
import MachineCoordinates from '@/components/machines/MachineCoordinates.vue'

describe('MachineCoordinates component', () => {
  test('is a Vue instance', () => {
    const wrapper = shallowMount(MachineCoordinates);

    expect(wrapper.isVueInstance()).toBeTruthy()
  });

  test('renders correctly', () => {
    const wrapper = shallowMount(MachineCoordinates);

    expect(wrapper.element).toMatchSnapshot()
  })
});
