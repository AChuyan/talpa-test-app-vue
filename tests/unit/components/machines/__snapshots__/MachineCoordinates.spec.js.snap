// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MachineCoordinates component renders correctly 1`] = `
<p
  class="machine-coordinates"
>
  <span>
    lat=unknown
  </span>
   
  <span>
    lng=unknown
  </span>
</p>
`;
